packages/medusa/src/services/product-category.ts

```diff
@@ -253,6 +261,38 @@ class ProductCategoryService extends TransactionBaseService {
       )
     })
   }
+
+  /**
+   * Accepts an input object and transforms product_category_id
+   * into product_category entity.
+   * @param productCategoryInput - params used to create/update
+   * @param {repository} A repository to query entity from
+   * @return transformed productCategoryInput
+   */
+  protected async transformParentIdToEntity_(
+    productCategoryInput:
+      | CreateProductCategoryInput
+      | UpdateProductCategoryInput,
+    repository: typeof ProductCategoryRepository
+  ): Promise<CreateProductCategoryInput | UpdateProductCategoryInput> {
+    // Typeorm only updates mpath when the category entity of the parent
+    // is passed into create/save. For this reason, everytime we create a
+    // category, we must fetch the entity and push to create
+    const parentCategoryId = productCategoryInput.parent_category_id
+
+    if (!parentCategoryId) {
+      return productCategoryInput
+    }
+
+    const parentCategory = await repository.findOne(
```

Comment:
suggestion(non-blocking): Use `this.retrieve` to get the "existence check" for free